Sources/xcprojectlint-package/CheckForWhitespaceSpecifications.swift

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2018 American Express Travel Related Services Company, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import Foundation
+
+public func checkForWhiteSpaceSpecifications(_ project: Project, errorReporter: ErrorReporter) -> Int32 {
+  var errors: [String] = []
+  var scriptResult = EX_OK
+
+  for namedGroup in project.groups.keys {
```

Comment:
Should we generate separate error for each whitespace specification? 
I think it will be very helpful when you have multiple specifications in a single group or file. Otherwise, removing one specification will result in the same error as before removal and may give false impression, that nothing has changed.

You should also be able to iterate over `values` to avoid `guard let group = project.groups[namedGroup] else { continue }`

Reply:
I started to do that, but think it would result in a ton of spam on the line.  Already, any group or file that has at least one will notify you, and the README tells the user what is checked, so they could look for them.  I guess, if that is too much back-and-forth for someone new to the tool, or getting an update, then we could fill the screen of each instance instead... I am not sure on which is preferred.

Reply:
Yeah, I think it's fine. Let's try out unified error message first and see if it's good enough.

I'd still ask you to iterate over `project.groups.values` to avoid `guard let group = project.groups[namedGroup] else { continue }`.
Same can be done with `project.fileReferences.values`